crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,449 @@
+    /// - **false**: Check only `private` properties. This is the default behavior.
+    ///   - Example:
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "checkAllProperties": false
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```typescript,use_options,diagnostic
+    /// class Example {
+    ///     public constantValue = 42;
+    ///     protected initializedInConstructor: string;
+    ///     // only `privateField` can be marked as readonly
+    ///     private privateField = true;
+    ///
+    ///     constructor(initializedInConstructor: string) {
+    ///         this.initializedInConstructor = initializedInConstructor;
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// - **false**: Check all properties (`public`, `protected`, `private` or none).
+    ///
+    ///   - Example:
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "checkAllProperties": true
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```typescript,use_options,diagnostic
+    /// class Example {
+    ///     // All properties below can be marked as readonly
+    ///     public constantValue = 42;
+    ///     protected initializedInConstructor: string;
+    ///     private privateField = true;
+    ///
+    ///     constructor(initializedInConstructor: string) {
+    ///         this.initializedInConstructor = initializedInConstructor;
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub UseReadonlyClassProperties {
+        version: "1.0.0",
+        name: "useReadonlyClassProperties",
+        language: "ts",
+        sources: &[RuleSource::EslintTypeScript("prefer-readonly")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseReadonlyClassProperties {
+    type Query = Ast<JsClassDeclaration>;
+    type State = TextRangeWithText;
+    type Signals = Box<[Self::State]>;
+    type Options = ReadonlyClassPropertiesOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let root = ctx.query();
+        let members = root.members();
+        let private_only = !ctx.options().check_all_properties;
+        let assignments = get_all_property_reassignments(&members);
+
+        let assignment_names = assignments.into_iter().filter_map(|assignment| {
+            assignment
+                .as_js_name()
+                .map(|name| name.to_trimmed_text())
+                .or_else(|| {
+                    assignment
+                        .as_js_private_name()
+                        .map(|private_name| private_name.to_trimmed_text())
+                })
+        });
+
+        let constructor_params = filter_mutable_constructor_parameters(root, private_only)
+            .into_iter()
+            .flat_map(|params| params.into_iter());
+
+        let mutable_class_property_members = members
+            .iter()
+            .filter_map(|member| filter_mutable_class_property(&member, private_only));
```

Comment:
Same here. Since this function is used only once, why doesn't it accept `members` and return the information that you need?